src/devices/generic_rc_power_socket.c

```diff
@@ -0,0 +1,162 @@
+/* Generic remote controlled power sockets using PT2262/PT2272 (or equivalent) protocol.
+ *
+ * This is called 'generic' because there are so many clones of this particular device and this code 
+ * will work with many of them, but still keep in mind not all of them are compatible even if they
+ * are built around PT2262/PT2272 or similar ICs because product designers sometimes get creative and use 
+ * a different allocation of the address/data bits of the 2262 chip or a different timing resistor. 
+ * The potential differences are usually around the way the DIP switches are used in the design. There can
+ * be differences in the actual number of bits allocated for remote id and the socket id fields, and there
+ * can also be differences in the order of bits in these fields. In most cases designers use 2-state DIP
+ * switches instead of tri-state so not all 3 allowable protocol states (0, 1, Z) are actually used. Also,
+ * the meaning of the DIP switches is not necessarily 0/1 as you would probably assume, instead they could
+ * be inverted or a 0/1 DIP bit value could be mapped to a Z transmitted value instead, and finally, there
+ * can be differences in how the on/off command is sent. Implementations usually use two bits but some might
+ * use only one. 
+ *
+ * The TH 111 device uses 2-state DIP switches, 5 for remote id mapped to A0-A4 (all can be set or unset) 
+ * and 5 for socket id mapped to A5, D5, D4, D3, D2 (only one can be set). The on/off command is sent via
+ * two individual bits (D1, D0). The mapping of the DIP switches and button values to actuall transmitted values 
+ * is not the intuitive one, so here we go:
+ *     Remote id: 0 -> Z, 
+ *                1 -> 1
+ *
+ *     Socket id: 1 -> 0, 
+ *                0 -> Z
+ *
+ *     Command:   On -> D1=Z,D0=0,
+ *                Off -> D1=0,D0=Z
+ *
+ * NOTE: this protcol uses 25 bit packets and might clash with other protocols using same packet length, notably other 
+ * 2262/2272 decoders and also seems to be similar enough to 'WS Temperature Sensor' leading to annoying duplicate 
+ * matches. Please disable those protocols manually to avoid such annoyances.
+ *
+ * Tested devices:
+ * - Somogyi Elektronic TH 111 (with HX2262/HX2272) (see https://www.somogyi.hu/product/taviranyithato-halozati-aljzat-szett-1db-aljzat-1db-taviranyito-th-111-12121)
+ *   Product pictures indicate strongly this might be a clone of the (yet untested) Avidsen indoor remote-controlled socket (see http://www.avidsen.com/product-sav?product_id=312&lang=en_US)
+ *
+ * Copyright (C) 2018 Adrian Nistor
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include "decoder.h"
+
+static int generic_rc_power_socket_callback(r_device *decoder, bitbuffer_t *bitbuffer)
```

Comment:
This function returns 0 for broken packets. The packets emitted by the remote when multiple buttons are pressed, either both On/Off for same socket or for multiple sockets, are considered invalid because the socket also seems to reject them and does not act on them.